src/connection/services/Vcard.ts

```diff
@@ -17,11 +18,129 @@ export default class Vcard extends AbstractService {
+              }
+
+              let photo = vcard.children('PHOTO');
+              if (avatar!=null&&mimetype!=null)
+              {
+                  let type;
+                  let binval;
+                  if (photo.length===0)
+                  {//XML BECAUSE OF CASE SENSIVITY
+                      photo = $($.parseXML('<PHOTO/>').documentElement);
+                      type = $($.parseXML('<TYPE/>').documentElement);
+                      binval = $($.parseXML('<BINVAL/>').documentElement);
+                      photo.append(type);
+                      photo.append(binval);
+                      vcard.append(photo);
+                  }
+                  else
+                  {
+                      type = photo.children('TYPE');
+                      binval = photo.children('BINVAL');
+                  }
+                  type.text(mimetype);
+                  binval.text(avatar);
+              }
+              else
+              {
+                  if (photo.length>0)
+                  {
+                      vcard.children('PHOTO').remove();
+                  }
+              }
+
+              return this.sendVCard(jid,avatar, vcard, getVcardIQ);
+          }
+          else
+          {
+              return false;
+          }
+      });
+   }
+
+   private sendVCard(jid : IJID, avatar: string, vcard, getVcardIQ): Promise<any>
+   {
+       let sendVcardIQ = $iq({
+                 type: 'set',
+                 to: jid.bare
+              }).cnode(vcard[0]);
+
+        //second send the new vCard with avatar to server
+        return this.sendIQ(sendVcardIQ).then((result)=>{
+          if ($(result).attr('type')==='result')
+          {
+            //load the vcard again, because there could be a resize method on the server and we would have another hash
+            return this.sendIQ(getVcardIQ).then((stanza) => {
+                if ($(stanza).attr('type')==='result')
+                {
+                      let vcard = $(stanza).find('vCard');
+
+                      vcard = $($.parseXML(vcard.get(0).outerHTML)).find('>vCard');
+
+                      if (!vcard.length) {
+                         //XML BECAUSE OF CASE SENSIVITY
+                         vcard = $($.parseXML('<vCard '+NS.get('VCARD')+'/>').documentElement);
+                      }
+
+                      if (!vcard.length)
+                         return false;
+
+                      let photo = vcard.children('PHOTO');
+                      if (photo!=null)
+                      {
+                           //NOW SEND THE PRESENCE
+                           return this.sendPresence(photo.find('BINVAL').text(), jid);
+                      }
+                }
+                else
+                    return false;
+            });
+          }
+          else
+            return false;
+      });
+   }
+
+   private sendPresence(avatar, jid: IJID) {
```

Comment:
Looks like duplicated code. Use `sendPresence` from `AbstractConnection` or even better trigger a resend, because we also want to include the status.

Reply:
problem is that we dont now if there is a additional method to shrink the images on server... so the photo will become a new hash. so i reload the vCard with the saved image and then i take this to send the presence hash